filebeat/input/filestream/input.go

```diff
@@ -323,19 +332,31 @@ func (inp *filestream) readFromSource(
 				log.Debugf("EOF has been reached. Closing.")
 			} else {
 				log.Errorf("Read line error: %v", err)
+				inp.metrics.ProcessingErrors.Inc()
 			}
+			inp.metrics.FilesClosed.Inc()
+
 			return nil
 		}
 
 		s.Offset += int64(message.Bytes)
 
+		inp.metrics.MessagesRead.Inc()
 		if message.IsEmpty() || inp.isDroppedLine(log, string(message.Content)) {
 			continue
 		}
 
+		inp.metrics.BytesProcessed.Add(uint64(message.Bytes))
+		inp.metrics.EventsInFlight.Inc()
+
 		if err := p.Publish(message.ToEvent(), s); err != nil {
+			inp.metrics.ProcessingErrors.Inc()
 			return err
 		}
+
+		inp.metrics.EventsInFlight.Dec()
```

Comment:
@faec, good point. I'm definitely open to suggestions on either renaming this value, eliminating it, or trying to track it another way as you mentioned, as long as it still pertains to filestream. If the internal queue already has a metric for this, and if filestream has nothing more to do with that event, would it still be worth tracking the event from a filestream metric? Perhaps hearing more about the internals of filestream would help us here.

Reply:
I guess it depends what exactly you're trying to measure -- there are various metrics in the pipeline for pending events once they enter the queue, but they won't be specific to filestream, they'll include all events that are being ingested. Once an event enters the queue, it is not directly handled by filestream anymore, but the filestream input still listens for it to be ingested upstream before advancing the position in the source file. If you want to measure how many outstanding events are still being tracked by filestream then you might want to use `TrackingCounter` when creating the publishing client -- see https://github.com/elastic/beats/blob/5a24ace1a526b6bd5015a2a47e87fd101695c96c/x-pack/filebeat/input/shipper/input.go#L172 for an example.

Reply:
I decided to remove that particular metric for now. While it may be nice to have in the future, it's a bit too intrusive to add at the moment, not to mention none of the other metrics that we've added recently have this sort of metric. Since we already have metrics for pending events in the queue, it's not critical that a filestream-specific one be added right now.